### PR TITLE
✨🗑️♻️🐛 Adds parameters in catalog for meta-modeling (⚠️ devops) and extras

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

First components for metamodeling

- ✨Catalog defines parameter nodes:  ``/parameter/number``, ``/parameter/boolean`` and ``/parameter/integer``
    - front-end service catalog is now moved to ``models_library.frontend_services_catalog`` and imported into catalog and director2 services
-  🗑️Deprecates ``displayOrder`` in input/outputs
-  ✨ Adds dev-feature flags in catalog and web-server
     -  ⚠️ devops: set ``CATALOG_DEV_FEATURES_ENABLED=1`` in master deployment
-  ♻️ Cleanup ``webserver.projects`` app module
     - improves request error handling
     - moves part of ``project_handlers.py`` into ``project_handlers_nodes.py`` to make file smaller
-  EXTRA: 🐛  ``synchronise_meta_data_table`` deletes table in chunks to improve PR #2428 that currently fails for large tables (>200k entries)

## Related issue/s

#2392 implementing parameters

## How to test

- in `.env` set ``CATALOG_DEV_FEATURES_ENABLED=1``
- ``make build up``
- open ``services_metadata table`` and ``service_access_rights``
image
image

Testing for changes
```command line
cd services/catalog
make install-dev
make test-dev
cd ...
cd packages/models_library
make install-dev
make test-dev

```

### CODE REVIEW

File: services/director-v2/tests/unit/test_db_comp_tasks.py [hunk 1]
```diff
@@ -49,12 +49,18 @@ def all_frontend_services(
...
-            service_key_version
-        )
-        if service_docker_data:
-            assert service_docker_data.key == "simcore/services/frontend/file-picker"
-            assert service_docker_data.outputs["outFile"].property_type == "data:*/*"
+    # FIXME: PC->SAN: this is ONLY for filepicker??
+    # It should also apply for other front-end functions, right?
+    for frontend_service_key in all_frontend_services:
+
+        if frontend_service_key == "simcore/services/frontend/nodes-group/macros/":
```
Comment (pcrespov) [inline services/director-v2/tests/unit/test_db_comp_tasks.py:56]: @odeimaiz should we also add ``simcore/services/frontend/nodes-group/macros/`` in the catalog?

File: services/director-v2/tests/unit/test_db_comp_tasks.py [hunk 2]
```diff
@@ -49,12 +49,18 @@ def all_frontend_services(
...
-    for frontend_service in all_frontend_services:
-        service_key_version = ServiceKeyVersion(key=frontend_service, version="24.34.5")
-        # check that it does not assert
-        service_docker_data: ServiceDockerData = _get_fake_service_details(
-            service_key_version
-        )
-        if service_docker_data:
-            assert service_docker_data.key == "simcore/services/frontend/file-picker"
-            assert service_docker_data.outputs["outFile"].property_type == "data:*/*"
+    # FIXME: PC->SAN: this is ONLY for filepicker??
```
Comment (pcrespov) [inline services/director-v2/tests/unit/test_db_comp_tasks.py:52]: @sanderegg based on the test name, is this really ONLY for the file-picker or can we use for all?

File: services/director-v2/tests/unit/test_db_comp_tasks.py (hunk 1 above)
Comment (odeimaiz) [reply services/director-v2/tests/unit/test_db_comp_tasks.py:56]: I would add it under development services

**Review by odeimaiz — APPROVED** [review]
Tested and working like a charm 💪

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py [hunk 3]
```diff
@@ -27,34 +23,23 @@
...
+# Examples are nodes like file-picker or parameter/*
+#
+_FRONTEND_SERVICES_CATALOG: Dict[str, ServiceDockerData] = {
+    meta.key: meta
+    for meta in iter_service_docker_data()
+    if any(name in meta.key for name in ["file-picker", "parameter"])
+}
+assert (  # nosec
+    len(_FRONTEND_SERVICES_CATALOG) == 4
+), "If this fails, review filter above and update"  # nosec
```
Comment (mguidon) [inline services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py:42]: This is a bit weird. Would only trigger if the service names match "file-picker" or "parameter". Or is that the intention? And what about the node grouping service?

**Review by mguidon — APPROVED** [review]

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py (hunk 3 above)
Comment (pcrespov) [reply services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py:42]: If somebody adds more front-end services in the catalog, probably the filter I use (which is not very solid) will cause troubles.

this assert intends to warn developer to review that filter again and update it to the new situation or even refactor it.

**Pull request merged**